lib/internal/main/test_runner.js

```diff
@@ -102,53 +100,41 @@ function filterExecArgv(arg) {
 }
 
 function runTestFile(path) {
-  return test(path, () => {
-    return new Promise((resolve, reject) => {
-      const args = ArrayPrototypeFilter(process.execArgv, filterExecArgv);
-      ArrayPrototypePush(args, path);
-
-      const child = spawn(process.execPath, args);
-      // TODO(cjihrig): Implement a TAP parser to read the child's stdout
-      // instead of just displaying it all if the child fails.
-      let stdout = '';
-      let stderr = '';
-      let err;
-
-      child.on('error', (error) => {
-        err = error;
-      });
-
-      child.stdout.setEncoding('utf8');
-      child.stderr.setEncoding('utf8');
-
-      child.stdout.on('data', (chunk) => {
-        stdout += chunk;
-      });
-
-      child.stderr.on('data', (chunk) => {
-        stderr += chunk;
-      });
-
-      child.once('exit', async (code, signal) => {
-        if (code !== 0 || signal !== null) {
-          if (!err) {
-            await PromiseAll(new SafeArrayIterator([finished(child.stderr), finished(child.stdout)]));
-            err = new ERR_TEST_FAILURE('test failed', kSubtestsFailed);
-            err.exitCode = code;
-            err.signal = signal;
-            err.stdout = stdout;
-            err.stderr = stderr;
-            // The stack will not be useful since the failures came from tests
-            // in a child process.
-            err.stack = undefined;
-          }
-
-          return reject(err);
-        }
-
-        resolve();
-      });
+  return test(path, async (t) => {
+    const args = ArrayPrototypeFilter(process.execArgv, filterExecArgv);
+    ArrayPrototypePush(args, path);
+
+    const child = spawn(process.execPath, args, { signal: t.signal });
```

Comment:
We could pass an `encoding` option and remove the `setEncoding` calls below.
```suggestion
    const child = spawn(process.execPath, args, { signal: t.signal, encoding: 'utf8' });
```